main.rb

```diff
@@ -0,0 +1,44 @@
+# define a method which takes in an array as its parameter
+# looping through the array
+# set the value of swap to false
+# if the lenght of array is 1 or less than 1 we shouldn't swap
+# comparison of first and second elements
+# If the element on the left is greater than the one on the right swap the two
+
+def bubble_sort(array)
+  array_length = array.size
+  return array if array_length <= 1
+
+  loop do
+    swap = false
+    (array_length - 1).times do |i|
+      if array[i] > array[i + 1]
+        array[i], array[i + 1] = array[i + 1], array[i]
+        swap = true
+      end
+    end
+    break unless swap
+  end
+  array
+end
+
+puts bubble_sort([5, 8, 3, 2, 1])
+
+def bubble_sort_by(array)
+  unsort = true
+  while unsort
+    i = 0
+    unsort = false
+    while i < (array.length - 1)
+      if (yield array[i], array[i + 1]).to_i >= 0
+        array[i], array[i + 1] = array[i + 1], array[i]
+        unsort = true
+      end
+      i += 1
+    end
+  end
+  print array
```

Comment:
Please make the `bubble_sort_by` method **return** the sorted array instead of `print`ing it.
  - [ ] Kindly remove the `print` keyword and make the method return the `array`.

Here is the "why": In the later lessons (and in your job), you will be learning about testing. When are testing your methods, you will need your method to **return** something so that you can test that returned value with the expected result. But when you `print` the sorted array, the returned value of that method becomes `nil`, which is not what we are looking for. 

A function takes input and returns/gives and output after processing the given input and then we see the output on the console after we call the method like this:

```ruby
print method(input)
```